maketeams3.py

```diff
@@ -104,31 +112,91 @@ def split_into_equal_groups_by_rating(players, group_number):
         last += avg
     return players_split
 
+
 def get_rating_bounds_of_split(split):
     min_ratings = [min([p.rating for p in board]) for board in split]
     max_ratings = [max([p.rating for p in board]) for board in split]
     min_ratings[-1] = 0
     max_ratings[0] = 5000
     return list(zip(min_ratings, max_ratings))
 
+
+def total_happiness(teams):
+    return sum([team.team_pref_score for team in teams])
+
+
+def team_rating_range(teams):
+    means = [team.getMean() for team in teams]
+    return max(means) - min(means)
+
+
+def squared_diff(a, b):
+    return (a - b)**2
+
+
+def variance(mean, xs):
+    return sum([squared_diff(mean, x) for x in xs]) / len(xs)
+
+
+def team_rating_variance(teams, league_mean=None):
+    if not league_mean:
+        league_mean = sum([team.getMean() for team in teams]) / len(teams)
+    return variance(league_mean, [team.getMean() for team in teams])
+
+
 @click.command()
 @click.option('--output', default="readable", type=click.Choice(['json', 'readable']))
 @click.option('--players', help='the json file containing the players.', required=True)
 @click.option('--boards', default=6, help='number of boards per team.')
 @click.option('--balance', default=0.8, help='proportion of all players that will be full time')
-def make_teams(players, output, boards, balance):
-    # input file is JSON data with the following keys: rating, name, in_slack, account_status, date_created, prefers_alt, friends, avoid, has_20_games.
+@click.option('--count', default=100, help='Number of iterations to fun happiness optimizer')
+def run(players, output, boards, balance, count):
+    player_data = get_player_data(players)
+
+    leagues = [make_league(player_data, boards, balance) for _ in range(count)]
+
+    max_happiness = max([total_happiness(l['teams']) for l in leagues])
+    happy_leagues = [l for l in leagues if total_happiness(l['teams']) == max_happiness]
+    print(f"{len(happy_leagues)} leagues of happiness {max_happiness} found")
+
+    for i, league in enumerate(happy_leagues):
```

Comment:
Any reason to output all and not just the lowest rating range one?

Reply:
I agree. I don't see the point in printing out all of them except for debugging purposes.

Reply:
This part is all up to you.  I have no idea what kind of output you'll truly find useful.  Just treat this part of the code as an example and configure it how you feel best.